File: rds/src/main/java/com/truenine/component/rds/properties/SnowflakeProperties.java
package com.truenine.component.rds.properties;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;

/**
 * @author TrueNine
 * @since 2023-04-01
 */
@Data
@ConfigurationProperties(prefix = "component.rds.snowflake")
public class SnowflakeProperties {
  Long workId = (long) 1;
  Long dataCenterId = (long) 2;
  Long sequence = (long) 3;
  Long startTimeStamp = (long) 100000;
}

File: rds/src/main/kotlin/com/truenine/component/rds/properties/SnowflakeProperties.kt
package com.truenine.component.rds.properties

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "component.snowflake")
data class SnowflakeProperties(
  var workId: Long = 1,
  var dataCenterId: Long = 2,
  var sequence: Long = 3,
  var startTimeStamp: Long = 100000
)
